aptos-move/framework/aptos-framework/sources/permissioned_signer.move

```diff
@@ -0,0 +1,661 @@
+        115792089237316195423570985008687907853269984665640564039457584007913129639935;
+
+    struct GrantedPermissionHandles has key {
+        active_handles: vector<address>
+    }
+
+    enum PermissionedHandle {
+        V1 {
+            master_account_addr: address,
+            permissions_storage_addr: address
+        }
+    }
+
+    enum StorablePermissionedHandle has store {
+        V1 {
+            master_account_addr: address,
+            permissions_storage_addr: address,
+            expiration_time: u64
+        }
+    }
+
+    enum PermissionStorage has key {
+        V1 {
+            perms: SimpleMap<Any, StoredPermission>
+        }
+    }
+
+    enum StoredPermission has store, copy, drop {
+        Unlimited,
+        Capacity(u256),
+    }
+
+    enum Permission<K> {
+        V1 {
+            owner_address: address,
+            key: K,
+            perm: StoredPermission,
+        }
+    }
+
+    /// Create an ephermeral permission handle based on the master signer.
+    ///
+    /// This handle can be used to derive a signer that can be used in the context of
+    /// the current transaction.
+    public fun create_permissioned_handle(master: &signer): PermissionedHandle {
+        assert_master_signer(master);
+        let permissions_storage_addr = generate_auid_address();
+        let master_account_addr = signer::address_of(master);
+
+        move_to(
+            &create_signer(permissions_storage_addr),
+            PermissionStorage::V1 { perms: simple_map::new() }
+        );
+
+        PermissionedHandle::V1 { master_account_addr, permissions_storage_addr }
+    }
+
+    /// Create an storable permission handle based on the master signer.
+    ///
+    /// This handle can be used to derive a signer that can be stored by a smart contract.
+    /// This is as dangerous as key delegation, thus it remains public(package) for now.
+    ///
+    /// The caller should check if `expiration_time` is not too far in the future.
+    public(package) fun create_storable_permissioned_handle(
+        master: &signer, expiration_time: u64
+    ): StorablePermissionedHandle acquires GrantedPermissionHandles {
+        assert_master_signer(master);
+        let permissions_storage_addr = generate_auid_address();
+        let master_account_addr = signer::address_of(master);
+
+        assert!(
+            timestamp::now_seconds() < expiration_time,
+            error::permission_denied(E_PERMISSION_EXPIRED)
+        );
+
+        if (!exists<GrantedPermissionHandles>(master_account_addr)) {
+            move_to<GrantedPermissionHandles>(
+                master, GrantedPermissionHandles { active_handles: vector::empty() }
+            );
+        };
+
+        vector::push_back(
+            &mut borrow_global_mut<GrantedPermissionHandles>(master_account_addr).active_handles,
+            permissions_storage_addr
+        );
+
+        move_to(
+            &create_signer(permissions_storage_addr),
```

Comment:
I'm a little confused from this line.
How do you create a signer from an address?
How is this safe?

How does this prevent me from doing
```move
move_to(
  &create_signer(victim_addr),
  SpamResource { ... }
);
```
or possibly worse?